modin/engines/ray/cudf_on_ray/frame/data.py

```diff
@@ -0,0 +1,275 @@
+            The column labels to extract.
+        col_numeric_idx : list of int
+            The column indices to extract.
+
+        Returns
+        -------
+        BasePandasFrame
+             A new BasePandasFrame from the mask provided.
+        """
+        if isinstance(row_numeric_idx, slice) and (
+            row_numeric_idx == slice(None) or row_numeric_idx == slice(0, None)
+        ):
+            row_numeric_idx = None
+        if isinstance(col_numeric_idx, slice) and (
+            col_numeric_idx == slice(None) or col_numeric_idx == slice(0, None)
+        ):
+            col_numeric_idx = None
+        if (
+            row_indices is None
+            and row_numeric_idx is None
+            and col_indices is None
+            and col_numeric_idx is None
+        ):
+            return self.copy()
+        if row_indices is not None:
+            row_numeric_idx = self.index.get_indexer_for(row_indices)
+        if row_numeric_idx is not None:
+            row_partitions_list = self._get_dict_of_block_index(0, row_numeric_idx)
+            if isinstance(row_numeric_idx, slice):
+                # Row lengths for slice are calculated as the length of the slice
+                # on the partition. Often this will be the same length as the current
+                # length, but sometimes it is different, thus the extra calculation.
+                new_row_lengths = [
+                    len(range(*idx.indices(self._row_lengths[p])))
+                    for p, idx in row_partitions_list.items()
+                ]
+                # Use the slice to calculate the new row index
+                new_index = self.index[row_numeric_idx]
+            else:
+                new_row_lengths = [len(idx) for _, idx in row_partitions_list.items()]
+                new_index = self.index[sorted(row_numeric_idx)]
+        else:
+            row_partitions_list = {
+                i: slice(None) for i in range(len(self._row_lengths))
+            }
+            new_row_lengths = self._row_lengths
+            new_index = self.index
+
+        if col_indices is not None:
+            col_numeric_idx = self.columns.get_indexer_for(col_indices)
+        if col_numeric_idx is not None:
+            col_partitions_list = self._get_dict_of_block_index(1, col_numeric_idx)
+            if isinstance(col_numeric_idx, slice):
+                # Column widths for slice are calculated as the length of the slice
+                # on the partition. Often this will be the same length as the current
+                # length, but sometimes it is different, thus the extra calculation.
+                new_col_widths = [
+                    len(range(*idx.indices(self._column_widths[p])))
+                    for p, idx in col_partitions_list.items()
+                ]
+                # Use the slice to calculate the new columns
+                new_columns = self.columns[col_numeric_idx]
+                assert sum(new_col_widths) == len(
```

Comment:
the code around here is really similar but not identical to what we have in PandasDataFrame.mask (now take_2d_labels_or_positional).  are the differences intentional?  or could we share some of it?